### PR TITLE
fix data race on TestGracefulShutdown

### LINKED ISSUE #9: testing: data race on TestGracefulShutdown
**Describe the bug**
https://github.com/knative/eventing/pull/4333/checks?check_run_id=1270178847

 **Expected behavior**
no data race

 **To Reproduce**
Steps to reproduce the behavior.

 **Knative release version**

 **Additional context**
Add any other context about the problem here such as proposed priority

/assign

### PR DESCRIPTION
**Pull request by runzexia**
Signed-off-by: runzexia <root@junwuhui.cn>

Fixes #9

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: pkg/adapter/receive_adapter_test.go [hunk 1]
```diff
@@ -156,21 +157,24 @@ var testCases = []testCase{
...
+	done := make(chan struct{})
+	go func(ctx context.Context) {
 		err := ra.Start(ctx)
 		if err != nil {
-			t.Error(err)
+			t.Error("Unexpected error:", err)
 		}
-		cancel()
-	}()
+		close(done)
```
Comment (markusthoemmes) [inline pkg/adapter/receive_adapter_test.go:167]: It's usually good to `defer` this at the start of the goroutine to guarantee it'll be called.

File: pkg/adapter/receive_adapter_test.go [hunk 2]
```diff
@@ -156,21 +157,24 @@ var testCases = []testCase{
...
+	}(ctx)
 
 	cancel()
-	<-ctx.Done()
-}
 
+	select {
+	case <-time.After(2 * time.Second):
```
Comment (markusthoemmes) [inline pkg/adapter/receive_adapter_test.go:173]: Is this time check necessary?

File: pkg/adapter/receive_adapter_test.go (hunk 1 above)
Comment (runzexia) [reply pkg/adapter/receive_adapter_test.go]: good point!

File: pkg/adapter/receive_adapter_test.go (hunk 2 above)
Comment (runzexia) [reply pkg/adapter/receive_adapter_test.go]: no not necessary, i removed it

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Comment by markusthoemmes** [comment]
/hold

Whoops habit!

**Pull request merged**